### PR TITLE
Add integration tests for dataset

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Adds integration tests for dataset operations

## 🐾 Next steps

Implement more integration sets

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1201481007343159/1201671951990194/f) (_internal_)

## 🛡 What tests cover this?

- These **are** the tests

## ❓ How to test this?

- run `cargo test`

## 📹 Demo

image

### CODE REVIEW

File: packages/engine/tests/units/data/csv/access/data/dataset.csv [hunk 1]
```diff
@@ -0,0 +1,3 @@
+1,true,black
+2,false,red
+3,true,gold
```
Comment (TimDiekmann) [inline packages/engine/tests/units/data/csv/access/data/dataset.csv:3]:
```suggestion
3,true,gold

```

File: packages/engine/tests/units/data/csv/immutable/data/dataset.csv [hunk 2]
```diff
@@ -0,0 +1,3 @@
+1,true,black
+2,false,red
+3,true,gold
```
Comment (TimDiekmann) [inline packages/engine/tests/units/data/csv/immutable/data/dataset.csv:3]:
```suggestion
3,true,gold

```

File: packages/engine/tests/units/data/csv/immutable/src/behaviors/test.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Ensures a dataset is immutable
+ */
+const behavior = (state, context) => {
+  context.data()["dataset.csv"][0][0];
```
Comment (Alfred-Mountfield) [inline packages/engine/tests/units/data/csv/immutable/src/behaviors/test.js:5]: Are you supposed to be setting something here? I don't understand what this does

Comment (TimDiekmann) [reply packages/engine/tests/units/data/csv/immutable/src/behaviors/test.js:5]: Yes, while refactoring the test I forgot to assign a value.
```suggestion
  context.data()["dataset.csv"][0][0] = "4";
```

**Review by Alfred-Mountfield — APPROVED** [review]

**Pull request merged**